### PR TITLE
changed colors depending on transportation mode

### PR DESCRIPTION
**Pull request by NNLNoah**
image
image
image
image

This pull request adds comprehensive unit tests for the `RoutePolyline` component and refactors its color and dash pattern logic for maintainability and clarity. The most important changes are grouped below:

**Testing improvements:**

* Added a new test suite `RoutePolyline-test.tsx` that thoroughly tests rendering behavior, color, and dash pattern for all travel modes, including edge cases like null routes and null travel modes.

**Code refactoring and maintainability:**

* Extracted logic for determining polyline color into a new helper function `getPolylineColor`, centralizing the mapping from travel mode to color.
* Extracted logic for determining dash patterns into a new helper function `getLineDashPattern`, centralizing the mapping from travel mode to dash style.
* Updated the `RoutePolyline` component to use the new helper functions for `strokeColor` and `lineDashPattern` props, improving readability and maintainability.

### CODE REVIEW

**Comment by Shamsam** [comment]
Lovely. If you need to walk to get to the bus stop, what colour is the path for the walking part?

**Review by FrancoisMartinez — DISMISSED** [review]
tested with different transportation mode and it works, good job

**Review by Shamsam — DISMISSED** [review]

**Review by Shamsam — DISMISSED** [review]

**Review by FrancoisMartinez — DISMISSED** [review]

**Comment by FrancoisMartinez** [comment]
I think you overwrote the shuttle bus color. We should keep it concordia maroon

File: mobile/src/components/LocationScreen/RoutePolyline.tsx [hunk 1]
```diff
@@ -1,32 +1,155 @@
...
+ * Determines the color for a step based on vehicle type or travel mode.
+ */
+function getStepColor(step: StepInfo): string {
+  // If it's a transit step with vehicle details, color by vehicle type
+  if (step.transitDetails) {
+    const vehicleType = step.transitDetails.vehicleType?.toUpperCase();
+    switch (vehicleType) {
+      case "SUBWAY":
+      case "RAIL":
+        return "#FF6B35"; // Metro/Rail: Orange-red
```
Comment (FrancoisMartinez) [inline mobile/src/components/LocationScreen/RoutePolyline.tsx]: lets move this hardcoded color and the other ones to our COLORS const in mobile\src\constants\index.ts

File: mobile/src/types/Directions.ts [hunk 2]
```diff
@@ -95,6 +95,7 @@ export interface StepInfo {
   instruction: string;
   maneuver: string;
   coordinates: { latitude: number; longitude: number }[];
+  travelMode?: string; // e.g., "WALK", "BICYCLE", "TRANSIT"
```
Comment (FrancoisMartinez) [inline mobile/src/types/Directions.ts]: We already have a travelMode enum that we could use here

File: mobile/src/components/LocationScreen/RoutePolyline.tsx [hunk 3]
```diff
@@ -1,32 +1,155 @@
...
 interface RoutePolylineProps {
   readonly route: RouteInfo | null;
   readonly travelMode: TravelMode | null;
 }
 
 /**
- * Renders the navigation route polyline on the map.
+ * Determines the color for a step based on vehicle type or travel mode.
+ */
+function getStepColor(step: StepInfo): string {
```
Comment (FrancoisMartinez) [inline mobile/src/components/LocationScreen/RoutePolyline.tsx]: This could maybe get moved to a utils file

File: mobile/src/components/LocationScreen/RoutePolyline.tsx [hunk 4]
```diff
@@ -1,32 +1,155 @@
...
+  }
+
+  // Default: Dark blue for walking
+  return "#40509F";
+}
+
+/**
+ * Determines the line dash pattern for a step.
+ */
+function getStepLineDashPattern(step: StepInfo): number[] | undefined {
```
Comment (FrancoisMartinez) [inline mobile/src/components/LocationScreen/RoutePolyline.tsx]: lets move this to a utils file aswell

File: mobile/src/types/Directions.ts (hunk 2 above)
Comment (NNLNoah) [reply mobile/src/types/Directions.ts]: good catch. fixed

File: mobile/src/components/LocationScreen/RoutePolyline.tsx (hunk 1 above)
Comment (NNLNoah) [reply mobile/src/components/LocationScreen/RoutePolyline.tsx]: nice. done.

File: mobile/src/components/LocationScreen/RoutePolyline.tsx (hunk 4 above)
Comment (NNLNoah) [reply mobile/src/components/LocationScreen/RoutePolyline.tsx]: good idea. done

File: mobile/src/components/LocationScreen/RoutePolyline.tsx (hunk 3 above)
Comment (NNLNoah) [reply mobile/src/components/LocationScreen/RoutePolyline.tsx]: absolutely. done

**Review by Shamsam — APPROVED** [review]

**Review by qrs-programmer — APPROVED** [review]
